pkg/alertmanager/api.go

```diff
@@ -422,15 +422,9 @@ func validateReceiverHTTPConfig(cfg commoncfg.HTTPClientConfig) error {
 	if cfg.BearerTokenFile != "" {
 		return errPasswordFileNotAllowed
 	}
-	if cfg.ProxyURL.URL != nil {
-		return errProxyURLNotAllowed
-	}
 	if cfg.OAuth2 != nil && cfg.OAuth2.ClientSecretFile != "" {
 		return errOAuth2SecretFileNotAllowed
 	}
-	if cfg.OAuth2 != nil && cfg.OAuth2.ProxyURL.URL != nil {
```

Comment:
Earlier today I opened the PR https://github.com/grafana/mimir/pull/2076 to test the firewall works with `cfg.ProxyURL.URL`. Do think it's feasible to have a similar test on `cfg.OAuth2.ProxyURL.URL` too?